xen/arch/arm/coproc/coproc.c

```diff
@@ -267,6 +267,22 @@ bool_t coproc_is_attached_to_domain(struct domain *d, const char *path)
     return is_created;
 }
 
+/*
+ * To make a decision can this device be assigned to the IOMMU context.
+ * As we only have device-tree support for now treat potential coproc device
+ * as a device-tree device.
+ */
+bool_t coproc_is_protected(device_t *dev)
```

Comment:
This is an analogy with passthrough's "dt_device_is_protected". So, coproc_is_protected means that both requirements were met:
1. It is a coproc
2. It is protected by IOMMU.
I tend to leave "coproc_is_protected", but I agree that comment in code should be corrected to avoid confusing.